graphql_compiler/compiler/emit_sql.py

```diff
@@ -239,19 +338,182 @@ def start_global_operations(self):
+            self._ir.query_metadata_table.get_location_info(fold_scope_location).type.name
+        ].alias()
+        outer_vertex_alias = self._current_alias.alias()
+        join_descriptor = self.get_join_descriptor(fold_scope_location)
+
+        return inner_vertex_alias, outer_vertex_alias, join_descriptor
+
+    def get_fold_outputs(self, fold_scope_location, join_descriptor):
+        """Generate output columns for innermost fold scope and add them to active SQLFoldObject."""
+        outputs = []
+        # find outputs for this fold in self._all_folded_outputs and add to self._fold_outputs
+        if fold_scope_location.fold_path in self._all_folded_outputs:
+            for fold_output in self._all_folded_outputs[fold_scope_location.fold_path]:
+                # this is for folds with the same fold path but different query paths
+                if (fold_output.base_location, fold_output.fold_path) == (
+                        fold_scope_location.base_location, fold_scope_location.fold_path):
+                    if fold_output.field == '_x_count':
+                        raise NotImplementedError(u'_x_count not implemented in SQL')
+
+                    # making explicit label forces column to have label as opposed to anon_label
+                    intermediate_fold_output_name = 'fold_output_' + fold_output.field
+                    # add array aggregated intermediate output to self._fold_outputs
+                    outputs.append(
+                        sqlalchemy.func.array_agg(
+                            self._fold.inner_vertex_alias.c[fold_output.field]
+                        ).label(intermediate_fold_output_name)
+                    )
+
+        # the group by field is the unique identifier for the outer vertex
+        outputs.append(self._fold.outer_vertex_alias.c[join_descriptor.from_column])
+
+        return sorted(outputs, key=lambda x: x.name, reverse=True)
+
+    def fold(self, fold_scope_location):
+        """Begin execution of a Fold Block."""
+        # This method creates the SELECT clause and GROUP BY for the fold subquery.
+        #
+        # SELECT will contain OuterVertex.uuid, a unique identifier for the
+        # OuterVertex determined by the edge descriptor from the vertex immediately outside the fold
+        # to the inner vertex where the fold outputs are.
+        #
+        # NOTE: right now inner vertex denotes both the vertex which is folded on AND the vertex at
+        # which the outputs are placed. This is only accurate when there are no traversals inside of
+        # the fold. Once functionality for traversals inside folds is introduced the folded vertex
+        # will no longer necessarily be the one that has the outputs.
+        #
+        # SELECT will also contain an ARRAY_AGG for each columns labeled for output inside the fold
+        #
+        # SELECT will also contain a COUNT(*) if _x_count is referred to somewhere in the query
```

Comment:
Can you make it clear that this is a TODO since we don't have COUNT implemented yet?